File: samples_generated_code/sample_1/phi4/baseline_code_py/code_row_66.py
import os

# Create and write to the file
with open('secret.txt', 'w') as file:
    file.write('U2FsdGVkX1/R+WzJcxgvX/Iw==')

# Change the file permissions to restrictive (read/write for owner only)
os.chmod('secret.txt', 0o600)

File: samples_generated_code/sample_3/phi4/baseline_code_java/code_row_66.java
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;

public class SecretFileCreator {
    public static void main(String[] args) {
        String filePath = "secret.txt";
        String content = "U2FsdGVkX1/R+WzJcxgvX/Iw==";

        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath))) {
            writer.write(content);
        } catch (IOException e) {
            e.printStackTrace();
        }

        // Set restrictive permissions: read-only for owner, no access for group and others
        java.nio.file.Files.setPosixFilePermissions(
                java.nio.file.Paths.get(filePath),
                java.nio.file.attribute.PosixFilePermission[] {
                    java.nio.file.attribute.PosixFilePermission.OWNER_READ
                }
        );
    }
}